File: C++/Medium/WrongAnswer/number-of-subarrays-with-bounded-maximum_6.cpp
class Solution {
public:
    int numSubarrayBoundedMax(vector<int>& nums, int left, int right) {
        
        int res = 0;
        int l = 0, r = 0, n = nums.size();
        while(r < n) {
            if (nums[r] >= left && nums[r] <= right) {
                res += r - l + 1;
            } else if (nums[r] < left) {
                res += r - l;
            } else {
                l = r + 1;
            }
            r++;
        }
        return res;
    }
};

File: Python3/Medium/RuntimeError/number-of-subarrays-with-bounded-maximum_6.py
class Solution:
    def numSubarrayBoundedMax(self, nums: List[int], left: int, right: int) -> int:
        
        # nums = [2,1,4,3], left = 2, right = 3
        # [2], [2, 1], [3]
        # [2, 1, 4], [1, 4, 3], [4, 3]
        # [2, 1, 4, 3]
        # [1, 4], [4, 3]